Add HyperNova's NIMFS circuit

arnaucube opened the pull request:
Implements [HyperNova](https://eprint.iacr.org/2023/573.pdf) NIMFS verifier circuit, except for the [CycleFold](https://eprint.iacr.org/2023/1192.pdf) part.

resolves #50

dmpierre reviewed (dismissed):
Lgtm, sorry for the late review on this! Hope this wasn't blocking you too much :)

arnaucube commented:
Rebased to latest `main` branch changes and updated the code to adapt to it. @dmpierre if you can change the changes since your review (https://github.com/privacy-scaling-explorations/sonobe/pull/99/commits/4ab13097967a041b5d3e046eb7086d1b46a76013), it's minimal, but was needed to make it work since latest main branch updates ^^
